pennylane/devices/qubit/simulate.py

```diff
@@ -12,18 +12,16 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 """Simulate a quantum script."""
-from typing import Union
-
 # pylint: disable=protected-access
 import pennylane as qml
-from pennylane.typing import TensorLike
+from pennylane.typing import Result
 
 from .initialize_state import create_initial_state
 from .apply_operation import apply_operation
 from .measure import measure
 
 
-def simulate(circuit: qml.tape.QuantumScript) -> Union[tuple, TensorLike]:
+def simulate(circuit: qml.tape.QuantumScript) -> Result:
```

Comment:
Should this be `Union[Result, ResultBatch]`? The `circuit` could have multiple measurements in which case the return value would be a nested tuple.

Reply:
Maybe I need better names.  I intended `Result` to be the result of executing a single tape, where `ResultBatch` would be the result of executing a batch of tapes.